src/racer/ast.rs

```diff
@@ -437,10 +434,7 @@ impl<'c, 's, 'v> visit::Visitor<'v> for ExprTypeVisitor<'c, 's> {
         debug!("visit_expr {:?}", expr);
         //walk_expr(self, ex, e)
         match expr.node {
-            ExprKind::Unary(_, ref expr) => {
-                self.visit_expr(expr);
-            }
-            ExprKind::AddrOf(_, ref expr) => {
+            ExprKind::Unary(_, ref expr) | ExprKind::AddrOf(_, ref expr) => {
```

Comment:
Can this remain as one variant per line? Eg

```rust
ExprKind::Unary(_, ref expr) |
ExprKind::AddrOf(_, ref expr) => {
    // ...
}
```

I find this version to be more readable.